python/lance/data/convert/nuscenes.py

```diff
@@ -0,0 +1,267 @@
+            pa.field("sample_data_token", pa.string()),
+            pa.field("category_token", pa.string()),
+            pa.field("mask", mask_schema), # TODO: change to RLE once type is created 
+
+            # Category
+            pa.field("category_token_", pa.string()),
+            pa.field("category_name_", pa.string()),
+            pa.field("category_description_", pa.string())
+        ])
+
+        # Object Ann
+        object_ann_schema = pa.struct([
+            pa.field("token", pa.string()),
+            pa.field("sample_data_token", pa.string()),
+            pa.field("category_token", pa.string()),
+            pa.field("attribute_tokens", pa.list_(pa.string())),
+            pa.field("bbox", Box2dType()),
+            pa.field("mask", mask_schema), # TODO: change to RLE once type is created
+
+            # Category
+            pa.field("category_token_", pa.string()),
+            pa.field("category_name_", pa.string()),
+            pa.field("category_description_", pa.string())
+        ])
+
+        return pa.schema([
+            # Sample is our root table, we don't rename these fields
+            pa.field("token", pa.string()),
+            pa.field("timestamp", pa.int64()),
+            pa.field("log_token", pa.string()),
+            pa.field("key_camera_token", pa.string()),
+            
+            # Log
+            pa.field("log_token_", pa.string()),
+            pa.field("log_logfile_", pa.string()),
+            pa.field("log_vehicle_", pa.string()),
+            pa.field("log_date_captured_", pa.string()),
+            pa.field("log_location_", pa.string()),
+            
+            # Calibrated Sensor
+            pa.field("calibrated_sensor_token_", pa.string()),
+            pa.field("calibrated_sensor_sensor_token_", pa.string()),
+            pa.field("calibrated_sensor_translation_", pa.list_(pa.float32(), 3)),
+            pa.field("calibrated_sensor_rotation_", pa.list_(pa.float32(), 4)),
+            pa.field("calibrated_sensor_camera_intrinsic_", pa.list_(pa.list_(pa.float32(), 3))),
+            pa.field("calibrated_sensor_camera_distortion_", pa.list_(pa.float32())), # can be 5 or 6 length
+            
+            # Sensor
+            pa.field("sensor_token_", pa.string()),
+            pa.field("sensor_channel_", pa.string()),
+            pa.field("sensor_modality_", pa.string()),
+
+            # Sample Data
+            pa.field("sample_data_token_", pa.string()),
+            pa.field("sample_data_sample_token_", pa.string()),
+            pa.field("sample_data_ego_pose_token_", pa.string()),
+            pa.field("sample_data_calibrated_sensor_token_", pa.string()),
+            pa.field("sample_data_filename_", pa.string()),
+            pa.field("sample_data_fileformat_", pa.string()),
+            pa.field("sample_data_width_", pa.int32()),
+            pa.field("sample_data_height_", pa.int32()),
+            pa.field("sample_data_timestamp_", pa.int64()),
```

Comment:
Use a timestamp type?

btw, giving that these fields will be visiable in sql, should we get rid of the suffix `_` for column name?

Reply:
done